src/main.rs

```diff
@@ -69,16 +76,18 @@ fn main() {
     // Parse & Build an AST
     let mut parser = Parser::new(lexer);
 
-    let ast_root = match parser.parse() {
+    let mut ast_root = match parser.parse() {
```

Comment:
The AST is technically an IL and LLVM is also. I'm not convinced there is added benefit of another IL in between the two, but I suppose we will see as we flesh it out. Either way, not really relevant to this PR.